mijia/src/bluetooth/mod.rs

```diff
@@ -823,32 +808,26 @@ impl BluetoothSession {
 }
 
 fn get_service_data(
-    device_properties: &HashMap<String, Variant<Box<dyn RefArg>>>,
-) -> Option<HashMap<String, Vec<u8>>> {
+    device_properties: OrgBluezDevice1Properties,
+) -> Option<HashMap<Uuid, Vec<u8>>> {
     // UUIDs don't get populated until we connect. Use:
     //     "ServiceData": Variant(InternalDict { data: [
     //         ("0000fe95-0000-1000-8000-00805f9b34fb", Variant([48, 88, 91, 5, 1, 23, 33, 215, 56, 193, 164, 40, 1, 0])
     //     )], outer_sig: Signature("a{sv}") })
     // instead.
     Some(
         device_properties
-            .get("ServiceData")?
-            // Variant(...)
-            .as_iter()?
-            .next()?
-            // InternalDict(...)
-            .as_iter()?
-            .tuples::<(_, _)>()
-            .filter_map(|(k, v)| {
-                let k = k.as_str()?.into();
-                let v: Option<Vec<u8>> = v
-                    .box_clone()
-                    .as_static_inner(0)?
-                    .as_iter()?
-                    .map(|el| Some(el.as_u64()? as u8))
-                    .collect();
-                let v = v?;
-                Some((k, v))
+            .service_data()?
+            .iter()
+            .filter_map(|(k, v)| match Uuid::parse_str(k) {
+                Ok(uuid) => {
+                    let v = cast::<Vec<u8>>(&v.0)?;
```

Comment:
(optional) should we be noisier if this cast fails?

Reply:
Added a warning log.